### ISSUE TITLE
Group block: When in its initial setup state, it's not a container and can't be dragged into

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#49256]
When a group block is inserted empty, it comes with a setup state allowing you to choose between the default flow group, a row, and a stack variation. In this state, the group looks like a group container in the list view, but it isn't, so I cannot drag any blocks into this container. GIF showing this:

![drag into group](https://user-images.githubusercontent.com/1204802/226848592-dd3379dc-f8ea-4a0d-9f9f-811885d40203.gif)

Filing this as a bug, as I would expect dragging a block into the group, even in its setup state, to have the dragged block nest under it. 

Related: #39064

**Comment by richtabor** [comment]
100%. This seems an easy win.

**Comment by andrewserong** [comment]
I did a little digging into this one, and it's a little more complex than I realised at first glance. Just jotting down a few technical notes. Feel free to ignore these, but I thought I'd write them down in case it helps anyone else looking. If no-one beats me to it, I'll look at seeing if I can put up a fix next week!

* The list view calls `canInsertBlocks` [here](https://github.com/WordPress/gutenberg/blob/bf39600fae6bf0731b7d9f281627ff9a2fff5cec/packages/block-editor/src/components/list-view/use-list-view-drop-zone.js#L229) to determine whether to allow the dragged blocks as children.
* Then the `canInsertBlockTypeUnmemoized` function checks `getBlockListSettings` for the Group block [here](https://github.com/WordPress/gutenberg/blob/a43afc673255965d919803cb32923a82855c1de4/packages/block-editor/src/store/selectors.js#L1547). A Group block that is in its placeholder state has no entry in the block list settings.
* When a Group block is in its placeholder state, it doesn't render out the result from `useInnerBlocksProps`, which means that [InnerBlocks](https://github.com/WordPress/gutenberg/blob/fff6d811cef2944935dca3166a9936da5a335428/packages/block-editor/src/components/inner-blocks/index.js#L236) is never rendered.
* InnerBlocks is responsible for calling `useNestedSettingsUpdate` [here](https://github.com/WordPress/gutenberg/blob/fff6d811cef2944935dca3166a9936da5a335428/packages/block-editor/src/components/inner-blocks/index.js#L62) which calls `updateBlockListSettings` [here](https://github.com/WordPress/gutenberg/blob/1d4be43396a2da75e001587b71b976e9687b215e/packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js#L122). This is how a block that renders its inner blocks winds up with an entry in the block list settings.

So, long story short — I _think_ if we were to try moving the call to `useNestedSettingsUpdate` up so that it gets called when `useInnerBlocksProps` is used, instead of when `InnerBlocks` is rendered, then we might be able to get the block list settings updated for container blocks when they're inserted / in their placeholder state, rather than just when they already have inner blocks. That's my idea at the moment, anyway 🙂

**Comment by andrewserong** [comment]
I've got a fix up in #49361 — it works by ensuring that `useInnerBlocksProps` and its children are output when the block is in the placeholder state. Seems to work pretty well so far 🤞
